tests_integration/nativeapp/test_deploy.py

```diff
@@ -445,3 +440,202 @@ def test_nativeapp_deploy_path_with_no_mapping(
                 env=TEST_ENV,
             )
             assert result.exit_code == 0
+
+
+# Tests that specifying a path and pruning result in an error
+@pytest.mark.integration
+def test_nativeapp_deploy_rejectes_pruning_when_path_is_specified(
+    runner,
+    temporary_working_directory,
+):
+    project_name = "myapp"
+    project_dir = "app root"
+    result = runner.invoke_json(
+        ["app", "init", project_dir, "--name", project_name],
+        env=TEST_ENV,
+    )
+    assert result.exit_code == 0
+
+    with pushd(Path(os.getcwd(), project_dir)):
```

Comment:
Nit, fixture `temporary_working_directory` already returns a path, so you should be able to use it directly instead of `os.getcwd()`.

Reply:
I can make that change, but then I'll do it throughout the file. I was just keeping with the pattern established by other tests.

Reply:
Tried it, some tests failed. I don't feel highly motivated to dig deeper :)